src/state/pools/helpers.ts

```diff
@@ -0,0 +1,24 @@
+import BigNumber from 'bignumber.js'
+import { Pool } from 'state/types'
+
+export const transformUserData = (userData: Pool['userData']) => {
+  return userData
+    ? {
+        allowance: new BigNumber(userData.allowance),
+        stakingTokenBalance: new BigNumber(userData.stakingTokenBalance),
+        stakedBalance: new BigNumber(userData.stakedBalance),
+        pendingReward: new BigNumber(userData.pendingReward),
+      }
+    : userData
+}
+
+export const transformPool = (pool: Pool): Pool => {
+  const { totalStaked, stakingLimit, userData, ...rest } = pool
+
+  return {
+    ...rest,
+    userData: transformUserData(userData),
+    totalStaked: new BigNumber(totalStaked),
+    stakingLimit: new BigNumber(stakingLimit),
+  } as Pool
+}
```

Comment:
While developing I noticed that pool data was incorrectly typed. They were typed as `BigNumber` but were actually coming back as strings. Nothing happened because we were casting the value as a `BigNumber` each time and/or we were not using the values a big numbers.